### DISCUSSION TITLE
Storybook doesn't work with .ts stories

### DISCUSSION THREAD

**Discussion by FatmaKahloun** [storybookjs/storybook#27360] [Help]
### Summary

Hi everyone, 

I'm developping a web component project using Lit, Panda css and Storybook. When I installed Storybook in my project it was installed with JS config files but I change the config files to TS also I change its default stories and all is good until then. But when I create my own story in TS (my-component.stories.ts) it doesn't work, I get no error but the component doesn't appear in the Storybook. But when I write the my own story in JS (my-component.stories.js) it works perfecttly. 

I didn't know the problem and I need to write stories using TS not JS, can anyone one have an idea ? 

Thank you a lot :smile:

**Comment by valentinpalkovic** [comment]
Can you please tell me how your .storybook/main.js looks like?

**Comment by FatmaKahloun** [comment]
there is my main.ts  file :

```
//main.ts 
import { StorybookConfig } from '@storybook/web-components-vite';

const config: StorybookConfig = {
  stories: ['../src/**/*.mdx', '../src/**/*.stories.@(js|jsx|mjs|ts|tsx)'],

  addons: [
    "@storybook/addon-links",
    "@storybook/addon-essentials",
    "@chromatic-com/storybook",
  ],
  framework: {
    name: "@storybook/web-components-vite",
    options: {},
  },
};
export default config;
```

And this is my preview.ts file : 

```
//preview.ts
import type { Preview } from '@storybook/web-components';
const preview: Preview = {
  parameters: {
    controls: {
      matchers: {
        color: /(background|color)$/i,
        date: /Date$/i,
      },
    },
  },
};

export default preview;

```
